src/proto_tcp.c

```diff
@@ -692,7 +692,13 @@ int tcp_bind_listener(struct listener *listener, char *errmsg, int errlen)
 		int defaultmss;
 		socklen_t len = sizeof(tmpmaxseg);
 
-		if (listener->rx.addr.ss_family == AF_INET)
+		int mptcp = listener->rx.proto->sock_prot == IPPROTO_MPTCP;
+
+		if (listener->rx.addr.ss_family == AF_INET && mptcp)
```

Comment:
it might be clearer to have:

```c
if (listener->rx.proto->sock_prot == IPPROTO_MPTCP) {
    if (v4)
        defaultmss = sock_inet_mptcp_maxseg_default;
    else
        (...)
} else {
    (...)
}
```

But up to you.

Reply:
seems indeed clearer like this